dali/pipeline/operator/builtin/split.h

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_PIPELINE_OPERATOR_BUILTIN_SPLIT_H_
+#define DALI_PIPELINE_OPERATOR_BUILTIN_SPLIT_H_
+
+#include <vector>
+
+#include "dali/pipeline/operator/operator.h"
+
+namespace dali {
+
+template <typename Backend>
+class Split : public Operator<Backend> {
+ public:
+  inline explicit Split(const OpSpec &spec) : Operator<Backend>(spec) {
+    DALI_ENFORCE(spec.HasTensorArgument("predicate"),
+                 "The 'predicate' argument is required to be present as argument input.");
+    RegisterTestsDiagnostics();
+  }
+
+  virtual inline ~Split() = default;
+
+  bool CanInferOutputs() const override {
+    return false;
+  }
+
+  bool SetupImpl(std::vector<OutputDesc> &output_desc, const Workspace &ws) override;
+  void RunImpl(Workspace &ws) override;
+
+  DISABLE_COPY_MOVE_ASSIGN(Split);
+
+ private:
+  void RegisterTestsDiagnostics();
+  void WriteTestsDiagnostics(const Workspace &ws);
+
+  USE_OPERATOR_MEMBERS();
+  static constexpr int kMaxCategories = 2;
+  std::array<int, kMaxCategories> category_counts_;
```

Comment:
I think that group or partition is a better name than category. It confused me quite a bit.

Reply:
Done